plugins/outputs/nats/nats.go

```diff
@@ -264,12 +264,22 @@ func (n *NATS) Write(metrics []telegraf.Metric) error {
 			n.Log.Debugf("Could not serialize metric: %v", err)
 			continue
 		}
-		// use the same Publish API for nats core and jetstream
-		err = n.conn.Publish(n.Subject, buf)
+		if n.Jetstream != nil {
+			_, err = n.jetstreamClient.PublishAsync(n.Subject, buf, jetstream.WithExpectStream(n.Jetstream.Name))
+		} else {
```

Comment:
Why do you change this to be async? This will only complicate the code... How about
```suggestion
		if n.Jetstream != nil {
			_, err = n.jetstreamClient.Publish(n.Subject, buf, jetstream.WithExpectStream(n.Jetstream.Name))
		} else {
```